src/libecalc/presentation/yaml/mappers/variables_mapper/time_series_collection_mapper.py

```diff
@@ -53,14 +54,22 @@ def _parse_date(date_input: Union[int, str]) -> datetime:
         return pandas.to_datetime(date_input, dayfirst=True).to_pydatetime()
 
 
-def _setup_time_series_data(
-    date_index: int,
-    time_series_resource: Resource,
-) -> Tuple[List[Union[datetime, pandas.Timestamp]], List[List]]:
-    time_vector = [_parse_date(date_input) for date_input in time_series_resource.data[date_index]]
-    columns = time_series_resource.data[:date_index] + time_series_resource.data[date_index + 1 :]
+def parse_time_vector(time_vector: List[str]) -> List[datetime]:
+    return [_parse_date(date_input) for date_input in time_vector]
 
-    return time_vector, columns
+
+def parse_time_series_from_resource(resource: Resource):
+    time_series_resource_headers = resource.get_headers()
+    if EcalcYamlKeywords.date in time_series_resource_headers:
+        # Find the column named "DATE" and use that as time vector
+        time_vector = resource.get_column(EcalcYamlKeywords.date)
+        headers = [header for header in time_series_resource_headers if header != EcalcYamlKeywords.date]
+    else:
+        # Legacy: support random names for time vector as long as it is the first column
```

Comment:
log a warning in order to catch this and possibly inform users?

Reply:
Should we? A resource with `Dates` as header instead of `Date` for example.